dao/mongodb.rya/src/main/java/mvm/rya/mongodb/instance/MongoDetailsAdapter.java

```
@@ -0,0 +1,143 @@
+        .add(PCJ_ENABLED_KEY, pcjDetails.isEnabled());
+        if(pcjDetails.getFluoDetails().isPresent()) {
+            pcjBuilder.add(PCJ_FLUO_KEY, pcjDetails.getFluoDetails().get().getUpdateName());
+        }
+        final List<DBObject> pcjDetailList = new ArrayList<>();
+        for(final PCJDetails pcjDetail : pcjDetails.getPCJDetails()) {
+            final BasicDBObjectBuilder indBuilbder = BasicDBObjectBuilder.start()
+            .add(PCJ_ID_KEY, pcjDetail.getId())
+            .add(PCJ_UPDATE_STRAT_KEY, pcjDetail.getUpdateStrategy().name());
+            if(pcjDetail.getLastUpdateTime().isPresent()) {
+                indBuilbder.add(PCJ_LAST_UPDATE_KEY, pcjDetail.getLastUpdateTime().get());
+            }
+            pcjDetailList.add(indBuilbder.get());
+        }
+        pcjBuilder.add(PCJ_PCJS_KEY, pcjDetailList.toArray());
+
+        final BasicDBObjectBuilder builder = BasicDBObjectBuilder.start()
+        .add(INSTANCE_KEY, details.getRyaInstanceName())
+        .add(VERSION_KEY, details.getRyaVersion())
+        .add(ENTITY_DETAILS_KEY, details.getEntityCentricIndexDetails().isEnabled())
+        .add(GEO_DETAILS_KEY, details.getGeoIndexDetails().isEnabled())
+        .add(PCJ_DETAILS_KEY, pcjBuilder.get())
+        .add(TEMPORAL_DETAILS_KEY, details.getTemporalIndexDetails().isEnabled())
+        .add(FREETEXT_DETAILS_KEY, details.getFreeTextIndexDetails().isEnabled());
+        if(details.getProspectorDetails().getLastUpdated().isPresent()) {
+            builder.add(PROSPECTOR_DETAILS_KEY, details.getProspectorDetails().getLastUpdated().get());
+        }
+        if(details.getJoinSelectivityDetails().getLastUpdated().isPresent()) {
+            builder.add(JOIN_SELECTIVITY_DETAILS_KEY, details.getJoinSelectivityDetails().getLastUpdated().get());
+        }
+        return (BasicDBObject) builder.get();
+    }
+
+    public static RyaDetails deserialize(final DBObject mongoObj) {
```

This should throw a checked exception if the object that is being deserialized is formatted incorrectly.